File: src/features/ai-platform/infra/commands/usage.ts
import { invoke } from '@tauri-apps/api/core';

export interface ProviderUsage {
  provider: string;
  cost: number;
  tokens: number;
  requests: number;
}

export interface ModelUsage {
  model: string;
  cost: number;
  tokens: number;
  requests: number;
}

export interface DailyUsage {
  date: string;
  cost: number;
  tokens: number;
  requests: number;
}

export interface ModelPricing {
  model: string;
  inputPricePerMillion: number;
  outputPricePerMillion: number;
  currency: string;
}

export interface UsageSummary {
  totalCost: number;
  totalTokens: number;
  totalRequests: number;
  byProvider: ProviderUsage[];
  byModel: ModelUsage[];
  dailyTrend: DailyUsage[];
  customPricing: ModelPricing[];
  days: number;
  source: string;
}

export function getAiPlatformUsageSummary(days: number) {
  return invoke<UsageSummary>('ai_platform_get_usage_summary', { days });
}

export function clearAiPlatformUsageRecords() {
  return invoke('ai_platform_clear_usage_records');
}

File: src-tauri/src/ai_platform/domain/usage.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UsageRecord {
	pub id: String,
	pub timestamp: i64,
	pub provider: String,
	pub model: String,
	pub tool: String,
	pub input_tokens: u64,
	pub output_tokens: u64,
	pub total_tokens: u64,
	pub cost: f64,
	pub currency: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ModelPricing {
	pub model: String,
	pub input_price_per_million: f64,
	pub output_price_per_million: f64,
	pub currency: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProviderUsage {
	pub provider: String,
	pub cost: f64,
	pub tokens: u64,
	pub requests: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ModelUsage {
	pub model: String,
	pub cost: f64,
	pub tokens: u64,
	pub requests: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DailyUsage {
	pub date: String,
	pub cost: f64,
	pub tokens: u64,
	pub requests: usize,
}